backend/benefit/applications/tests/test_application_search.py

```diff
@@ -30,19 +44,25 @@ def setup_application_data(application, archived):
 @pytest.mark.parametrize(
     "q, detected_pattern",
     [
-        ("HEL 2024-000123", SearchPattern.AHJO),
-        ("12345", SearchPattern.NUMBERS),
-        ("0877830-1", SearchPattern.NUMBERS),
-        ("040337W935P", SearchPattern.SSN),
+        ("HEL 2024-123456", SearchPattern.AHJO),
+        ("54321", SearchPattern.NUMBERS),
+        ("0877830-3", SearchPattern.NUMBERS),
+        ("010101W123P", SearchPattern.SSN),
         ("No result Oy", SearchPattern.COMPANY),
         (
             "No result Oy nimi:nobody",
             f"{SearchPattern.ALL} {SearchPattern.IN_MEMORY}-fallback",
         ),
     ],
 )
-def test_search_with_no_results(handler_api_client, q, detected_pattern):
-    response = handler_api_client.get(f"/v1/search/?q={q}")
+def test_search_with_no_results(handler_api_client, q, detected_pattern, application):
+    application = setup_application_data(application, False)
+    params = urlencode(
+        {
+            "q": q,
+        },
+    )
+    response = handler_api_client.get(f"{api_url}?{params}")
```

Comment:
This is somewhat better than the last PR, I suppose. I was thinking of something like `get_batch_detail_url` in the application batch tests, though, where the parameters go through the django/DRF API, too, instead of string concatenation.